daemon/workit.cpp

```diff
@@ -501,11 +522,13 @@ int work_it(CompileJob &j, unsigned int job_stat[], MsgChannel *client, CompileR
                     }
 
                     kill(pid, SIGTERM); // Most likely crashed anyway ...
-                    return_value = EXIT_COMPILER_CRASHED;
-                    client_fd = -1;
-                    input_complete = true;
+                    if (input_complete) {
+                        return_value = EXIT_COMPILER_CRASHED;
+                    }
                     delete fcmsg;
                     fcmsg = 0;
+                    close(sock_in[1]);
+                    sock_in[1] = -1;
```

Comment:
bah, this makes the code _less_ readable imo. -1 is a very well established magic number for invalid file descriptors, being the error indicator value of almost all related stdlib functions. also, it's much quicker to read.